src/storage/project.ts

```diff
@@ -186,13 +182,13 @@ async function renameOrCopyProject(
  */
 export async function deleteProject(
     storage: commonStorage.Storage, projectName: string): Promise<void> {
-  const modulePaths: string[] = await storage.listFilePaths(
-      storageNames.makeModulePathRegexPattern(projectName));
+  const projectFileNames: string[] = await storage.list(
+      storageNames.makeProjectDirectoryPath(projectName));
 
-  for (const modulePath of modulePaths) {
-    await storage.deleteFile(modulePath);
+  for (const projectFileName of projectFileNames) {
```

Comment:
Should this logic be above the storage or in the storage?   It might be simpler for the storage to know when you are deleting a directory that you mean to delete everything in it.

Reply:
Done.